Adaptors/RabbitMQ/src/QueueMessageHandler.cs

```diff
@@ -68,22 +68,25 @@ public ValueTask DisposeAsync()
       case QueueMessageStatus.Waiting:
         /* Negative acknowledging this message will send it
          to the retry exchange, see PullQueueStorage.cs */
-        channel_.BasicNack(basicGetResult_.DeliveryTag,
-                           false,
-                           true);
+        (await channel_.GetConnectionAsync(CancellationToken)
+                       .ConfigureAwait(false)).BasicNack(basicGetResult_.DeliveryTag,
```

Comment:
You can split this statement into 2, and move the GetConnectionAsync out of the switch.